test/nolib/AdmitDoesNotSimpl.fst

```diff
@@ -0,0 +1,36 @@
+module AdmitDoesNotSimpl
+
+#lang-pulse
+open Pulse.Nolib
+
+assume val foo : int -> slprop
+
+fn test (x y : int)
+  requires foo (fst (x,y))
+  ensures foo x
+{
+  admit(); (* Should show foo (fst (x,y)) *)
+}
+
+fn test2 (x y : int)
+  requires foo (fst (x,y))
+  ensures foo x
+{
+  assert foo x;
+  admit(); (* Should show foo x *)
+}
+
+fn test3 ()
+  requires foo (1+1)
+  ensures foo 2
+{
+  admit(); (* Should show foo (1+1) *)
+}
+
+fn test4 ()
+  requires foo (1+1)
+  ensures foo 2
+{
+  ();
+  admit(); (* Stil foo (1+1), but we could accept that an empty statement triggers normalization. *)
```

Comment:
I totally disagree with the proposal in the comment.  It is extremely confusing when adding an extra statement breaks unrelated code.  The statement `foo ()` should always behave the same as `(); foo ()`, `show_proof_state; foo ()`, `foo (); ()` (if `foo` returns unit), etc.

Reply:
I agree, what I was trying to say here is that we may want admit() to print the context before any of the eager normalization/elimination steps happen, and was thinking it could maybe always print an unprocessed context. Since the eager steps will happen for the `()`, it would be a way of printing the processed one. None of that should matter for non-admits though.

I anyway removed the comment.